MessengerOnFileForLinux/DisplayChat/src/ConsoleWindow.cpp

```diff
@@ -26,34 +26,62 @@ void ConsoleWindow::displayRegistrationWindow()
     move(0,0);
     printw("Hello ");
     printw(getenv("USER"));
-    printw(". You are using messenger first time. You need create account.");
+    printw(". You are using messenger first time. You need create account. Enter your password twice.");
     move(2,0);
+    printw("Enter the password : ");
     refresh();
 }
 
-void ConsoleWindow::displayLoggedWindow()
+void ConsoleWindow::displaySignInWindow()
 {
     clear();
     move(0,0);
     printw("Hello ");
     printw(getenv("USER"));
     printw(". If you want use messenger, you need to log in.");
     move(2,0);
+    printw("Enter the password : ");
     refresh();
 }
 
 std::string ConsoleWindow::getStringFromConsoleWindow()
 {
-    std::string input;
-    nocbreak();
-    echo();
-    int ch = getch();
-    while ( ch != '\n' )
+    std::string output;
+    char enteredCharacter{};
+    while (enteredCharacter != '\n')
     {
-        input.push_back( ch );
-        ch = getch();
+        constexpr int firstAsciiCharacter = 0;
+        constexpr int lastAsciiCharacter = 127;
+        constexpr int delInAsciiCode = 127;
+
+        enteredCharacter = getch();
+
+        if (firstAsciiCharacter <= enteredCharacter and
```

Comment:
w std jest coś co sprawdza czy mieści się w zakresie, nie pamiętam teraz nazwy. Ewentualnie nawet funkcje: czyZnak

Reply:
jest, ale czytałem ze jest deprecated. Mam juz to opakowane w funkcje w lokalnym commicie

Reply:
na pewno jest nowy, który nie jest deprecated

Reply:
No ja nic nie znalazłem

Reply:
boost::irange - to nie będzie przypadkiem to?

Reply:
albo to:
https://stackoverflow.com/questions/17095324/fastest-way-to-determine-if-an-integer-is-between-two-integers-inclusive-with
Choć jeśli pierwsze działa to wolę to z boosta

Reply:
z tego co wiem to range służa do iteracji , a nie określania zakresu. Wrzucam to do funkcji bez kombinowania. Prosty i najbardziej przejrzysty sposób